python/ray/tune/utils/log.py

```diff
@@ -0,0 +1,46 @@
+from enum import Enum
+from typing import Any, Callable, Union
+
+
+class Verbosity(Enum):
+    MINIMAL = 0
+    EXPERIMENT = 1
+    TRIAL_NORM = 2
+    TRIAL_DETAILS = 3
+
+    def __int__(self):
+        return self.value
+
+
+verbosity: Union[int, Verbosity] = Verbosity.TRIAL_DETAILS
+
+
+def set_verbosity(level: Union[int, Verbosity]):
+    global verbosity
+
+    if isinstance(level, int):
+        verbosity = Verbosity(level)
+    else:
+        verbosity = verbosity
+
+
+def verbose_log(logger: Callable[[str], Any], level: Union[int, Verbosity],
+                message: str):
+    """Log `message` if specified level exceeds global verbosity level.
+
+    `logger` should be a Callable, e.g. `logger.info`. It can also be
+    `print` or a logger method of any other level - or any callable that
+    accepts a string.
+    """
+    if has_verbosity(level):
+        logger(message)
```

Comment:
By passing the logger this should not happen, right? I.e. in `one.py` I instantiate a `logger`, which will log as `DEBUG one.py ...` - and if I call `verbose_log(logger.info, ...)` it will use the same logger and thus print the correct file. Right?

I thought about using custom loglevels for this but it seems to be much overhead and might interfere with other parts of Ray. Overall it seems introducing custom log levels is bad practice.

Reply:
Hmm, I think verbose_log will not use the same logger: 

```
2020-11-09 20:05:56,650	INFO log.py:36 -- Total run time: 2.39 seconds (2.35 seconds for the tuning loop).
```

Reply:
(which instead should show tune.py)

Reply:
I ran:  python examples/async_hyperband_example.py --smoke-test

Reply:
Sorry, this almost slipped. You're right, and that's because of our logger format `"%(asctime)s\t%(levelname)s %(filename)s:%(lineno)s -- %(message)s"` - I thought it was `name` instead of `filename`. 

Since we currently used it only in one location anyway I removed that utility function. It should be fine now.